### DISCUSSION TITLE
【PLONK】关于 Quotient poly 分解

### DISCUSSION THREAD

**Discussion by Demian101** [z2o-k7e/halo2-step-by-step#57] [Q&A] [answered]
<img width="706" alt="image" src="https://github.com/zkp-co-learning/halo2-step-by-step/assets/33189338/cdda3f58-ddd9-4760-a88f-c7aac92335ab">

背景： 在 Scroll 的[这篇文章](https://scroll.io/blog/proofGeneration#user-content-fnref-2) 中，如上图，提到了商多项式分解，以降低 degree 从而达到不推高 KZG setup degree 的目的。

我的问题是：考虑这样一个 Q 多项式:

$$
Q(x) = 1 + x+ x^2 + x^3 + ... + x^{19} 
$$

即使把 Q(x) 分解为 两个较小的多项式  $Q_{lo}(x), \ \  Q_{hi}(x)$

比如：

$$
Q_{lo}(x)= 1 + x+ x^2 + x^3 + ... + x^{11} , 
$$

$$
Q_{hi}(x)= x^{12} +  x^{13} + x^{14} + ... + x^{19}
$$

感觉 $Q_{hi}(x)$ 的 degree  还是 19 ，并没有减少？ 对嘛？

$Q_{hi}(x) \cdot x^n$ ，其 degree 又变成了 19+n

我觉得是我哪里理解错了，但是看了一阵也没看出来 🍭

**Comment by Pupil1999** [comment]
$Q_{hi}(x)=x^3+x^4+x^4+\cdots +x^{10}$，是除以 $x^{n}$ 之后的结果

**Comment by kurtpan666** [comment, marked as answer]
For your case, $n=10$:

$Q_{\text{lo}}(x) =Q_{\text{hi}}(x)= 1+ x+\ldots + x^{9}$

Only requires about half of the original degree.
